src/js/components/DataTable/TableHeaderCell.jsx

```diff
@@ -20,6 +24,27 @@ const TableHeaderCell = ({
     onKeyPress: () => {},
   };
 
+  if (tooltip) {
+    return (
+      <CustomTooltip content={tooltipLabel}>
+        <div
+          {...(sortable ? sortableProps : {})}
+          style={style}
+          className={`rt-th ${className} ${dynamicClassName?.(columnId)}`}
+        >
+          {children}
+          {required && <span className="ml-1 required">&#42;</span>}
+          {sortable && (
+            <div className="sorting-arrows">
+              <RiArrowUpSFill className="arrow-up" />
+              <RiArrowDownSFill className="arrow-down" />
+            </div>
+          )}
+        </div>
```

Comment:
let's wrap lines 30-43 in a separate component since it's used also below. By doing so we won't have to remember to refactor two places.